src/math.rs

```diff
@@ -185,25 +185,22 @@ pub fn crt(r: &[i64], m: &[i64]) -> (i64, i64) {
 ///
 /// assert_eq!(math::floor_sum(6, 5, 4, 3), 13);
 /// ```
+#[allow(clippy::many_single_char_names)]
 pub fn floor_sum(n: i64, m: i64, mut a: i64, mut b: i64) -> i64 {
+    assert!((0..1i64 << 32).contains(&n));
+    assert!((1..1i64 << 32).contains(&m));
     let mut ans = 0;
```

Comment:
Let's wrap the variables with `std::num::Wrapping<u64>`.  Note that if you shadow the variables, you need to check if a and b are negative beforehand.